library/core/src/ptr/const_ptr.rs

```diff
@@ -60,44 +63,37 @@ impl<T: ?Sized> *const T {
 
     /// Casts a pointer to its raw bits.
     ///
-    /// This is equivalent to `as usize`, but is more specific to enhance readability.
-    /// The inverse method is [`from_bits`](#method.from_bits).
-    ///
-    /// In particular, `*p as usize` and `p as usize` will both compile for
-    /// pointers to numeric types but do very different things, so using this
-    /// helps emphasize that reading the bits was intentional.
-    ///
-    /// # Examples
+    /// In general, pointers cannot be understood as "just an integer"
+    /// and cannot be created from one without additional context.
     ///
-    /// ```
-    /// #![feature(ptr_to_from_bits)]
-    /// let array = [13, 42];
-    /// let p0: *const i32 = &array[0];
-    /// assert_eq!(<*const _>::from_bits(p0.to_bits()), p0);
-    /// let p1: *const i32 = &array[1];
-    /// assert_eq!(p1.to_bits() - p0.to_bits(), 4);
-    /// ```
+    /// If you would like to treat a pointer like an integer anyway,
+    /// see [`addr`][] and [`with_addr`][] for the responsible way to do that.
     #[unstable(feature = "ptr_to_from_bits", issue = "91126")]
-    pub fn to_bits(self) -> usize
+    pub fn to_bits(self) -> [u8; core::mem::size_of::<*const ()>()]
     where
         T: Sized,
     {
-        self as usize
+        unsafe { core::mem::transmute(self) }
```

Comment:
Uuh I don't like this part.^^ See https://github.com/rust-lang/unsafe-code-guidelines/issues/286.

Reply:
You may notice I deprecated its twin -- in this form it is purely for "I really need intptr_t just as a way to talk about a pointer's bytes in an irreversible way" purposes. 

But yes I would just deprecate it, imo.

Reply:
Also IMO it should be `transmute(self.addr())` to avoid the sketchy ptr-to-int transmute.

Reply:
I have almost no experience with rust, so I may be mis-parsing this. However, it seems to me that this does indeed need to be `self` and not `self.addr()` if you want the raw pointer representation. For CHERI `self.addr()` will not give the all the pointer bits, only the 32/64 address bits without the metadata.

Reply:
Oh I see.
Hm but this produces `u8` with provenance on them which I don't think we want, either.^^